File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/api/io/exception/DefaultHttpException.java
package org.dvbviewer.controller.data.api.io.exception;

import java.io.IOException;

public class DefaultHttpException extends IOException {

    public DefaultHttpException(String url, Throwable cause) {
        super(cause.getMessage()
                + System.getProperty("line.separator")
                + System.getProperty("line.separator")
                + "when accessing url:"
                + System.getProperty("line.separator")
                + System.getProperty("line.separator")
                + url);
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/api/io/exception/DefaultHttpException.kt
package org.dvbviewer.controller.data.api.io.exception

import java.io.IOException

class DefaultHttpException(url: String, cause: Throwable) : IOException(cause.message
        + System.getProperty("line.separator")
        + System.getProperty("line.separator")
        + "when accessing url:"
        + System.getProperty("line.separator")
        + System.getProperty("line.separator")
        + url)
